update getting started with hosted/local setups (catering for miners adoption) (#216)
stratum-mining/stratumprotocol.org: src/getting-started.md

# Getting Started with running SRI - Stratum V2 reference implementation

This document aims to assist users in deploying the SRI software stack efficiently. Stratum v2 as a protocol is flexible, and allows users to run multiple different configurations.

There are 4 possible configurations for SV2 deployments:
- **Config A**: SV2 firmware with Job Declaration Protocol
- **Config B**: SV2 firmware without Job Declaration Protocol
- **Config C**: SV1 firmware without Job Declaration Protocol
- **Config D**: SV1 firmware with Job Declaration Protocol

The Job Declaration Protocol allows miners to construct their own block templates.

We are going to focus on Configs C and D, because most of the mining industry today still uses SV1 firmware.

> The SRI team is hosting a Pool, Template Provider (TP) and Job Declarator Server (JDS) to aid testing on the miner side.

Reviewer: ```suggestion
The SRI community is hosting a Pool, Template Provider (TP) and Job Declarator Server (JDS) to aid testing on the miner side.
```

Alternatively, you can also deploy your own Pool, TP and JDS to get the full infrastructure experience.

# Table of Contents
- [Prerequisites](#prerequisites)
  - [Install Rust](#install-rust)
  - [Clone SRI repository](#clone-sri-repository)
  - [Run Template Provider](#run-template-provider)
- [Choose your Setup](#choose-your-setup)
  - [Hosted Config C](#hosted-config-c)
  - [Local Config C](#local-config-c)
  - [Hosted Config D](#hosted-config-d)
  - [Local Config D](#local-config-d)
- [Connect Mining Devices](#connect-mining-devices)
- [CPU miner](#cpu-miner)
- [Adjust tproxy-config (optional)](#adjust-tproxy-config-optional)
- [Monitoring for blocks](#monitoring-for-blocks)
- [Customizing configuration](#customizing-configuration)
  - [Customize the `pool_signature` string](#customize-the-pool_signature-string)
  - [Customize the `coinbase_output` script](#customize-the-coinbase_output-script)

> - [Community Support](#xxx)

Reviewer: ```suggestion
- [Community Support](#community-support)
```

## Prerequisites

### Install Rust

> You'll needRust installed on your machine. If it's not:

Reviewer: ```suggestion
You'll need Rust installed on your machine. If it's not:
```

```bash
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```

### Clone SRI repository

```bash
git clone https://github.com/stratum-mining/stratum
```

### Run Template Provider

⚠️ Note: This is only necessary if you're going to be running either:
- Local Config C
- Hosted Config D
- Local Config D

You can skip this step if you're running Hosted Config C.

> Download a release from Sjors' fork of Bitcoin Core from https://github.com/Sjors/bitcoin/releases

Reviewer: ```suggestion
Download a release from Sjors' fork of Bitcoin Core from [https://github.com/Sjors/bitcoin/releases](https://github.com/Sjors/bitcoin/releases)
```

cd roles/translator/config-examples/
cargo run -- -c tproxy-config-local-jdc-example.toml
```

You're ready to start mining. Skip to the [Connect Mining Devices](#connect-mining-devices) step.

---

## Connect Mining Devices

Connect mining device - Translator Proxy will be running on port `34255`, so you will need just to edit your mining device(s) configuration, adding the line:

`stratum+tcp://<tProxy ip>:34255`

If you don't have an ASIC, skip to the [CPU miner step](#cpu-miner) below.

---

> ## CPU Miner

Reviewer: ```suggestion
### CPU Miner
```

If you don't have a physical miner, you can do tests with CPU Miner.

Setup the correct CPUMiner for your OS:
- You can download the binary directly from [here](https://sourceforge.net/projects/cpuminer/files/);
- Or compile it from [https://github.com/pooler/cpuminer](https://github.com/pooler/cpuminer)

On the CPUMiner directory:
 
```
./minerd -a sha256d -o stratum+tcp://localhost:34255 -q -D -P
```

> ## Adjust `tproxy-config` (optional)

Reviewer: ```suggestion
### Adjust `tproxy-config` (optional)
```

Depending on mining device you do run, you may have to adjust `tproxy-config*.toml` file in order to adjust the `min_individual_miner_hashrate` and `channel_nominal_hashrate` parameters to have an optimal difficulty target for the channel.

> 

Reviewer: ```suggestion
```
